libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -598,22 +597,84 @@ string YulUtilFunctions::clearStorageRangeFunction(Type const& _type)
 {
 	string functionName = "clear_storage_range_" + _type.identifier();
 
-	solUnimplementedAssert(_type.isValueType(), "...");
+	solAssert(_type.storageBytes() >= 32, "Expected smaller value for storage bytes");
 
 	return m_functionCollector->createFunction(functionName, [&]() {
 		return Whiskers(R"(
 			function <functionName>(start, end) {
-				for {} lt(start, end) { start := add(start, 1) }
+				for {} lt(start, end) { start := add(start, <increment>) }
 				{
-					sstore(start, 0)
+					<setToZero>(start, 0)
 				}
 			}
 		)")
 		("functionName", functionName)
+		("setToZero", storageSetToZeroFunction(_type))
+		("increment", _type.storageSize().str())
 		.render();
 	});
 }
 
+string YulUtilFunctions::clearStorageArrayFunction(ArrayType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+
+	if (_type.baseType()->storageBytes() < 32)
+	{
+		solAssert(_type.baseType()->isValueType(), "Invalid storage size for non-value type.");
+		solAssert(_type.baseType()->storageSize() <= 1, "Invalid storage size for type.");
+	}
+
+	if (_type.baseType()->isValueType())
+		solAssert(_type.baseType()->storageSize() <= 1, "Invalid size for value type.");
+
+	string functionName = "clear_storage_array_" + _type.identifier();
+
+	return m_functionCollector->createFunction(functionName, [&]() {
+		if (_type.isDynamicallySized())
+			return Whiskers(R"(
+				function <functionName>(slot) {
+					<clearDynamicArray>(slot)
```

Comment:
Why not implement it right here?

Reply:
I suppose we can do that now. The original implementation was called from another place (setToZero in LValue for StorageArrayLength). We instead call resizeArray with len 0 to do the same.

Reply:
I haven't done this yet as I was hoping for some clarification whether we want to keep using resizeArray with len 0 for setToZero in LValue StorageArrayLength or whether we want to call the clearDynamicArray function instead.

Reply:
I think it is best to implement both dynamic and static arrays directly here, maybe even in the same template. I think clearRange should be used for both.

The main reason is because resizing arrays will be removed at some point anyway, and we will only have
 - add one element
 - remove one element
 - fully delete